src/tests/class_tests/openms/source/AASequence_test.cpp

```diff
@@ -908,74 +917,122 @@ START_SECTION([EXTRA] Arbitrary tag in peptides using square brackets)
     TEST_EQUAL(test_other.size(), 8)
     TEST_STRING_SIMILAR(test_other.toString(), "PEPTN[1600.230654]IDE")
     TEST_STRING_SIMILAR(test_other.toUniModString(), "PEPTN[1600.230654]IDE")
-    TEST_STRING_SIMILAR(test_other.toBracketString(false), "PEPTN[1600.230654]IDE")
+    TEST_STRING_SIMILAR(test_other.toBracketString(false, false), "PEPTN[1600.230654]IDE")
+    TEST_STRING_SIMILAR(test_other.toBracketString(false, true), "PEPTN[+1486.1877258086]IDE")
     TEST_EQUAL(test_other.toUnmodifiedString(), "PEPTNIDE")
+    TEST_REAL_SIMILAR(test_other.getMonoWeight(), aa_original.getMonoWeight() + 1600.230654)
+    TEST_REAL_SIMILAR(test_other.getMonoWeight(), aa_withn.getMonoWeight() + 1486.1877258086)
 
-    TEST_STRING_SIMILAR(test_other.toString(), test_seq.toString()) // the peptides should be equal
+    TEST_EQUAL(test_other, test_seq) // the peptides should be equal
+
+    // test that we can re-read the string from BracketString
+    test_other = AASequence::fromString(test_seq.toBracketString(false, true));
+    TEST_EQUAL(test_other.size(), 8)
+    TEST_STRING_SIMILAR(test_other.toString(), "PEPTN[+1486.1877258086]IDE")
+    TEST_STRING_SIMILAR(test_other.toUniModString(), "PEPTN[1600.230654]IDE")
+    TEST_STRING_SIMILAR(test_other.toBracketString(false, false), "PEPTN[1600.230654]IDE")
+    TEST_STRING_SIMILAR(test_other.toBracketString(false, true), "PEPTN[+1486.1877258086]IDE")
+    TEST_EQUAL(test_other.toUnmodifiedString(), "PEPTNIDE")
+    TEST_REAL_SIMILAR(test_other.getMonoWeight(), aa_original.getMonoWeight() + 1600.230654)
+    TEST_REAL_SIMILAR(test_other.getMonoWeight(), aa_withn.getMonoWeight() + 1486.1877258086)
+
+    // TEST_STRING_SIMILAR(test_other.toString(), test_seq.toString()) // the peptides should be equal
   }
 
   // test N-terminal modification
   {
-    AASequence test_seq = AASequence::fromString("[1600.230654]IDE");
+    AASequence test_seq = AASequence::fromString(".[1601.2384790319]IDE");
     TEST_EQUAL(test_seq.size(), 3)
-    TEST_EQUAL(test_seq.toString(), ".[1600.230654]IDE")
-    TEST_STRING_SIMILAR(test_seq.toUniModString(), ".[1600.230654]IDE")
-    TEST_STRING_SIMILAR(test_seq.toBracketString(false), "n[1600.230654]IDE")
+    TEST_STRING_SIMILAR(test_seq.toString(), ".[1601.2384790319]IDE")
+    TEST_STRING_SIMILAR(test_seq.toUniModString(), ".[1601.2384790319]IDE")
+    TEST_STRING_SIMILAR(test_seq.toBracketString(false, false), "n[1601.2384790319]IDE") // an extra H for the N-terminus
+    TEST_STRING_SIMILAR(test_seq.toBracketString(false, true), "n[+1600.230654]IDE")
```

Comment:
this is what has been fixed here: `n[1601.2384790319]IDE` is now equivalent to `n[+1600.230654]IDE`